code/forms/SiteTreeURLSegmentField.php

```diff
@@ -30,7 +30,7 @@ public function getAttributes() {
 			parent::getAttributes(),
 			array(
 				'data-prefix' => $this->getURLPrefix(),
-				'data-suffix' => '?stage=Stage',
+				'data-suffix' => '?stage=' . $this->getURLSuffix(),
```

Comment:
Noted. Please see the conversation here for a suggestion on divergence from this implementation: https://github.com/silverstripe/silverstripe-cms/pull/1449